test/bigint.jl

```diff
@@ -401,3 +401,16 @@ end
 
 # Issue #24298
 @test mod(BigInt(6), UInt(5)) == mod(6, 5)
+
+@testset "cmp has values in [-1, 0, 1], issue #28780" begin
+    bigrand() = rand(-big(2)^rand(1:rand(1:1000)):big(2)^rand(1:rand(1:1000)))
+    for T in (Base.BitInteger_types..., BigInt, Float64, Float32, Float16, BigFloat)
+        @test cmp(big(2)^130, one(T)) == 1
+        @test cmp(-big(2)^130, one(T)) == -1
```

Comment:
How about testing these with `===` to ensure that the type matches as well?

Reply:
Ok, done. I removed the testing with `BigFloat` as this is handled in the `MPFR` module, and can return `Int32`. I can make a PR for `MPFR` if you think it's useful.